gobblin-yarn/src/main/java/gobblin/yarn/YarnHelixUtils.java

```diff
@@ -199,4 +206,48 @@ public static void addFileAsLocalResource(FileSystem fs, Path destFilePath, Loca
 
     return environmentVariableMap;
   }
-}
+
+  public static Optional<Integer> takePort(int desiredPort) {
```

Comment:
If the `desiredPort` is not available, is there a reason to sequentially increase the port value and check if each one is available. If the desiredPort is not available, I believe `new ServerSocket(0).getLocalPort()` will find an available port and return it.